ee/clickhouse/queries/enterprise_cohort_query.py

```diff
@@ -42,79 +43,71 @@ def check_negation_clause(prop: PropertyGroup) -> Tuple[bool, bool]:
 
 class EnterpriseCohortQuery(FOSSCohortQuery):
     def get_query(self) -> Tuple[str, Dict[str, Any]]:
-
         if not self._outer_property_groups:
             # everything is pushed down, no behavioral stuff to do
             # thus, use personQuery directly
             return self._person_query.get_query(prepend=self._cohort_pk)
 
         # TODO: clean up this kludge. Right now, get_conditions has to run first so that _fields is populated for _get_behavioral_subquery()
-        conditions, condition_params = self._get_conditions()
-        self.params.update(condition_params)
-
-        subq = []
-
-        if self.sequence_filters_to_query:
-            sequence_query, sequence_params, sequence_query_alias = self._get_sequence_query()
-            subq.append((sequence_query, sequence_query_alias))
-            self.params.update(sequence_params)
-        else:
-            behavior_subquery, behavior_subquery_params, behavior_query_alias = self._get_behavior_subquery()
-            subq.append((behavior_subquery, behavior_query_alias))
-            self.params.update(behavior_subquery_params)
-
-        person_query, person_params, person_query_alias = self._get_persons_query(prepend=str(self._cohort_pk))
-        subq.append((person_query, person_query_alias))
-        self.params.update(person_params)
+        conditions = self._get_conditions()
 
         # Since we can FULL OUTER JOIN, we may end up with pairs of uuids where one side is blank. Always try to choose the non blank ID
-        q, fields = self._build_sources(subq)
-
-        final_query = f"""
-        SELECT {fields} AS id  FROM
-        {q}
-        WHERE 1 = 1
-        {conditions}
-        """
-
-        return final_query, self.params
+        q, fields = self._build_sources(
+            [
+                (
+                    self._get_sequence_query() if self.sequence_filters_to_query else self._get_behavior_subquery(),
+                    self.FUNNEL_QUERY_ALIAS,
+                ),
+                (self._get_persons_query(prepend=str(self._cohort_pk)), self.PERSON_TABLE_ALIAS),
+            ]
+        )
 
-    def _get_condition_for_property(self, prop: Property, prepend: str, idx: int) -> Tuple[str, Dict[str, Any]]:
+        final_query = QueryFragment(
+            """
+            SELECT {fields} AS id
+            FROM {q}
+            WHERE 1 = 1
+            {conditions}
+            """,
+            fields=fields,
+            q=q,
+            conditions=conditions,
+        )
 
-        res: str = ""
-        params: Dict[str, Any] = {}
+        return final_query.sql, final_query.query_params
 
+    def _get_condition_for_property(self, prop: Property, prepend: str, idx: int) -> QueryFragment:
         if prop.type == "behavioral":
             if prop.value == "performed_event":
-                res, params = self.get_performed_event_condition(prop, prepend, idx)
+                return self.get_performed_event_condition(prop, prepend, idx)
```

Comment:
being able to see I can ignore the rest of this method once I've hit the branch I care about is 💯